api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -326,10 +326,128 @@ public CohortDefinition getPatientsWhoSuspendedTreatment() {
   }
 
   /** @return B8: Number of dead patients during the current month */
-  public CohortDefinition getPatientsWhoDied() {
-    EncounterWithCodedObsCohortDefinition cd = getStateOfStayCohort();
-    cd.addIncludeCodedValue(hivMetadata.getPatientHasDiedConcept());
-    cd.setName("Number of dead patients during the current month");
+  public CohortDefinition getPatientsWhoDied(Boolean data1) {
+
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.addParameter(new Parameter("onOrAfter", "onOrAfter", Date.class));
+    cd.addParameter(new Parameter("onOrBefore", "onOrBefore", Date.class));
+    cd.addParameter(new Parameter("locationList", "location", Location.class));
+    String sql =
+        "SELECT patient_id"
+            + "            FROM (SELECT patient_id, Max(death_date) AS death_date"
+            + "            FROM (SELECT p.patient_id, Max(e.encounter_datetime) AS death_date"
+            + "            FROM patient p "
+            + "                     JOIN encounter e "
+            + "                          ON p.patient_id = e.patient_id "
+            + "                     JOIN obs o "
+            + "                          ON e.encounter_id = o.encounter_id "
+            + "            WHERE p.voided = 0 "
+            + "              AND e.voided = 0 "
+            + "              AND e.location_id = :locationList "
+            + "              AND e.encounter_type = ${adultoSeguimento} ";
+    if (data1 == true) {
```

Comment:
The block could be simplified to:
```
if(data1) {
...
} else {
...
}

Reply:
Updated, value is required since is a parameter for the method.